packages/dds/tree/src/test/utils.ts

```diff
@@ -596,15 +597,22 @@ export function validateSnapshotConsistency(
 	idDifferentiator: string | undefined = undefined,
 ): void {
 	assert.deepEqual(
-		treeA.tree,
-		treeB.tree,
+		prepareTreeForCompare(treeA.tree),
+		prepareTreeForCompare(treeB.tree),
 		`Inconsistent document tree json representation: ${idDifferentiator}`,
 	);
+
 	// Note: removed trees are not currently GCed, which allows us to expect that all clients should share the same
 	// exact set of them. In the future, we will need to relax this expectation and only enforce that whenever two
 	// clients both have data for the same removed tree (as identified by the first two tuple entries), then they
 	// should be consistent about the content being stored (the third tuple entry).
-	const mapA = nestedMapFromFlatList(treeA.removed);
+	const mapA = nestedMapFromFlatList(
```

Comment:
Probably worth a comment explaining why we need to do something special for `treeA` here, but not `treeB` below.

Reply:
thats a bug.